pages/lessons/projects/6.mdx

```diff
@@ -0,0 +1,796 @@
+(**BAT**) are all smart contracts that follow a standard, the ERC20 standard.
+
+## Why do we want to create yet another Token?
+
+A lot of reasons, but here are some:
+
+- We are learning and we want to know how the ecosystem works
+- We have a project or DAO that needs some means of governance
+- We have a community that wants to be a part, take some ownership or support
+  our project
+- We need a token with extra functionality that doesn't exist yet
+
+The possibilities are endless.
+
+## What do we need to do with our Tokens?
+
+Some basic functionality that our tokens need to provide are:
+
+- Transfer tokens from one account to another
+- Get the current token balance for an account
+- Get the total supply of tokens available in the network
+- Approve an amount of tokens from an account to be spent by a third-party
+  account
+
+<SideDrawer buttonText="More info on the ERC20 Standard">
+
+To achive the functionality needed for an ERC20 token, the standard requires us
+to implement 9 methods and 2 events.
+
+Here is a quick overview of what our contracts needs to be called an ERC20 Token
+contract:
+
+```solidity
+// Methods:
+function name() public view returns (string)
+function symbol() public view returns (string)
+function decimals() public view returns (uint8)
+function totalSupply() public view returns (uint256)
+function balanceOf(address _owner) public view returns (uint256 balance)
+function transfer(address _to, uint256 _value) public returns (bool success)
+function transferFrom(address _from, address _to, uint256 _value) public returns (bool success)
+function approve(address _spender, uint256 _value) public returns (bool success)
+function allowance(address _owner, address _spender) public view returns (uint256 remaining)
+
+// Events:
+event Transfer(address indexed _from, address indexed _to, uint256 _value)
+event Approval(address indexed _owner, address indexed _spender, uint256 _value)
+```
+
+You can check the details in the full EIP-20 titled
+[ERC-20: Token Standard](https://eips.ethereum.org/EIPS/eip-20#token)
+
+</SideDrawer>
+
+## Has someone created a battle-tested library for the standard?
+
+Yes. OpenZeppelin (among others) has created an implementation we can inherit to
+create
+our own tokens easily.
+
+## Let's code, but first...
+
+Before we start hacking away without any structure to follow, what is our
+project's structure and what do we want to achieve?
+
+Since we are _BUIDLING_, i'd like to think of our project, or smart contract,
+the same way as we would _roughly_ build a house:
+
+    * 1 - Foundations
+    * 2 - Structure: Framing, Walls and Roof
+    * 3 - Doors and Windows
+    * 4 - Security
+    * 5 - Final touches
+
+Before we build, we need a stable, leveled surface to place our foundations so
+we can build on top.
+
+The tools for coding and the project's file/folder structure will be our
+foundations.
+
+## Setting up Foundations
+
```

Comment:
Not sure if this is necessary, but just a suggestion. Would it be helpful to mention the version of Node.js used? I got through the entire lesson successfully using Node 19.4.0. Not sure if an older version such as 8.0.0 would yield the same result.

Reply:
Yes. This is also worth mentioning!

Reply:
Well, this is the interesting part of using foundry... You don't need *Node.js* **at all** to do this lesson.
Do you recall using `npm`, `node`, `npx` or `yarn` commands during the lesson?

I'm gonna let that sink in... 😉